File: go/vehicle-purchase/vehicle_purchase.go
package purchase

import (
	"sort"
)

// NeedsLicense determines whether a license is needed to drive a type of vehicle. Only "car" and "truck" require a license.
func NeedsLicense(kind string) bool {
	if kind == "car" || kind == "truck" {
		return true
	}
	return false
}

// ChooseVehicle recommends a vehicle for selection. It always recommends the vehicle that comes first in lexicographical order.
func ChooseVehicle(option1, option2 string) string {
	stringSlice := [2]string{option1, option2}
	sort.Strings(stringSlice[:])
	return stringSlice[0] + " is clearly the better choice."
}

// CalculateResellPrice calculates how much a vehicle can resell for at a certain age.
func CalculateResellPrice(originalPrice, age float64) float64 {
	switch {
	case age < 3.0:
		return originalPrice * 0.80
	case age >= 10:
		return originalPrice * 0.50
	default:
		return originalPrice * 0.70
	}
}

File: cpp/vehicle-purchase/vehicle_purchase.cpp
#include "vehicle_purchase.h"

namespace vehicle_purchase {

    // needs_license determines whether a license is needed to drive a type of vehicle. Only "car" and "truck" require a license.
    bool needs_license(std::string kind){
        return kind == "car" || kind == "truck";
    }

    // choose_vehicle recommends a vehicle for selection. It always recommends the vehicle that comes first in lexicographical order.
    std::string choose_vehicle(std::string option1, std::string option2) {
        std::string suffix = " is clearly the better choice.";
        if (option1 < option2) return option1 + suffix;
        return option2 + suffix;
    }

    // calculate_resell_price calculates how much a vehicle can resell for at a certain age.
    double calculate_resell_price(double original_price, double age) {
        // TODO: Return the age-corrected resell price.
        if (age < 3) {
          return original_price * 0.80;
        } else if (age < 10) {
          return original_price * 0.70;
        } else {
          return original_price * 0.50;
        }
    }

}  // namespace vehicle_purchase